testsuite/tests/basic/boxedints.ml

```diff
@@ -72,6 +74,30 @@ struct
     test 4 (to_int (of_int 0x3FFFFFFF)) 0x3FFFFFFF;
     test 5 (to_int (of_int (-0x40000000))) (-0x40000000);
 
+    testing_function "unsigned_to_int";
+    test 1 (unsigned_to_int (of_int 0)) (Some 0);
+    test 2 (unsigned_to_int (of_int 123)) (Some 123);
+    test 3 (unsigned_to_int minus_one)
+      (match Sys.word_size with
+       | 32 -> None
+       | 64 -> Some (int_of_string "0xFFFFFFFF")
+       | _ -> assert false);
+    test 4 (unsigned_to_int max_int)
+      (match Sys.word_size with
+       | 32 -> None
+       | 64 -> Some (to_int max_int)
+       | _ -> assert false);
+    test 5 (unsigned_to_int min_int)
+      (match Sys.word_size with
+       | 32 -> None
+       | 64 -> Some (int_of_string "0x80000000")
+       | _ -> assert false);
+    test 6 (unsigned_to_int (of_int Stdlib.max_int))
```

Comment:
Reading this made me realize that the overflow behavior of `Int32.of_int` is not specified on 64-bit platforms.  Since the test depends on a truncation semantics (i.e. taking the result modulo 2^32), it might be worth documenting it.

Reply:
Thanks, added.